File: algos/217-containsDuplicate/217-containsDuplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        # Time Efficiency
        hashmap = set()

        for num in nums:
            if num in hashmap:
                return True
            hashmap.add(num)
        return False
        
        # # Space Efficiency
        # nums.sort()

        # for n in range(1, len(nums)):
        #     if nums[n] == nums[n-1]:
        #         return True
            
        # return False

File: algos/217-containsDuplicate/217-containsDuplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    const numsSet = new Set()

    for (let num of nums) {
        if (numsSet.has(num)) return true
        numsSet.add(num)
    }
    return false

    // // Cool Trick
    // // Create new set, check set length against length of nums
    // let numsSet = new Set(nums);
    // return (nums.length!=numsSet.size)
};